### PR TITLE
systemError: state refactor

### PR DESCRIPTION
**Pull request by sorja**
Committed with --no-verify with errors:

```
✖ eslint --cache --fix:
webapp/app/appErrors/appErrors.js
  15:21  error  'error' is missing in props validation                  react/prop-types
  15:28  error  'closeAppError' is missing in props validation          react/prop-types
  15:28  error  'closeAppError' is already declared in the upper scope  no-shadow
  21:7   error  Missing an explicit type attribute for button           react/button-has-type
  31:22  error  'errors' is missing in props validation                 react/prop-types
  31:30  error  'closeAppError' is missing in props validation          react/prop-types
  31:30  error  'closeAppError' is already declared in the upper scope  no-shadow
  33:13  error  'errors.map' is missing in props validation             react/prop-types

webapp/app/appErrorsMiddleware.js
  6:32  error  'axios' is already declared in the upper scope  no-shadow

webapp/store/system/index.js
   2:1  error  Dependency cycle via ./i18n:4=>./reducer:2=>@webapp/app/actions:3  import/no-cycle
  10:1  error  Dependency cycle via ./reducer:2=>@webapp/app/actions:3            import/no-cycle
  13:1  error  Dependency cycle via ./reducer:2=>@webapp/app/actions:3            import/no-cycle

webapp/store/system/reducer.js
  4:1  error  Dependency cycle via ./reducer:2=>@webapp/app/actions:3=>@webapp/store/system:8  import/no-cycle
  5:1  error  Dependency cycle via ./reducer:2=>@webapp/app/actions:3=>@webapp/store/system:8  import/no-cycle
```

### CODE REVIEW

File: webapp/app/appErrors/appErrors.js [hunk 1]
```diff
@@ -3,15 +3,14 @@ import './appErrors.scss'
...
 import * as R from 'ramda'
 
 import { useI18n } from '@webapp/store/system'
 
-import * as ErrorsState from './appErrorsState'
+import Markdown from '../../components/markdown'
 
-import { closeAppError } from './actions'
+import * as ErrorsState from '../../store/system/serviceError/state'
+import { closeAppError } from '../../store/system/serviceError/actions'
```
Comment (minotogna) [inline webapp/app/appErrors/appErrors.js:13]: this must be 
```import { ServiceErrorActions, useServiceError } from '@webapp/store/system'```

and useDispatch

Comment (ramirobg94) [inline webapp/app/appErrors/appErrors.js:13]: I think that we are doing in this way:

`import { ServiceSystemErrorState, ServiceSystemErrorActions } from '@webapp/store/system'`

`ServiceSystemErrorActions.closeAppError`

File: webapp/app/appErrors/appErrors.js [hunk 2]
```diff
@@ -31,15 +30,15 @@ const AppError = ({ error, closeAppError }) => {
...
         <AppError error={error} closeAppError={closeAppError} />
       </CSSTransition>
     ))}
   </TransitionGroup>
 )
 
-const mapStateToProps = state => ({
+const mapStateToProps = (state) => ({
   errors: ErrorsState.getAppErrors(state),
 })
```
Comment (ramirobg94) [inline webapp/app/appErrors/appErrors.js:43]: and replacing this by `useSystemServiceErrors()`

File: webapp/app/appErrorsMiddleware.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import axios from 'axios'
 import * as R from 'ramda'
 
-import { appErrorCreate } from './appErrors/actions'
+import { appErrorCreate } from '../store/system/serviceError/actions'
```
Comment (minotogna) [inline webapp/app/appErrorsMiddleware.js:4]: this must be 
```import { ServiceErrorActions } from '@webapp/store/system'```

File: webapp/store/system/serviceError/reducer.js [hunk 4]
```diff
@@ -1,8 +1,8 @@
-import { exportReducer } from '../../utils/reduxUtils'
+import { exportReducer } from '@webapp/utils/reduxUtils'
 
 import { appErrorCreate, appErrorDelete } from './actions'
 
-import * as ErrorsState from './appErrorsState'
+import * as ErrorsState from './state'
```
Comment (minotogna) [inline webapp/store/system/serviceError/reducer.js:5]: this must be 
```import * as ServiceErrorState from './state'```

File: webapp/store/system/serviceError/reducer.js [hunk 5]
```diff
@@ -1,8 +1,8 @@
...
 
 const actionHandlers = {
   [appErrorCreate]: (state, { error }) => ErrorsState.assocAppError(error)(state),
```
Comment (minotogna) [inline webapp/store/system/serviceError/reducer.js:8]: this must be ```[ServiceErrorActions.serviceErrorCreate]: ...```

File: webapp/app/appErrors/appErrors.js (hunk 2 above)
Comment (minotogna) [reply webapp/app/appErrors/appErrors.js:43]: better ```useServiceErrors```

File: webapp/app/appErrorsMiddleware.js [hunk 6]
```diff
@@ -1,12 +1,12 @@
...
-import { appErrorCreate } from './appErrors/actions'
+import { ServiceErrorActions } from '@webapp/store/system'
 
-const createAxiosMiddleware = (axios) => ({ dispatch }) => {
-  axios.interceptors.response.use(null, (error) => {
+const createAxiosMiddleware = (axiosInstance) => ({ dispatch }) => {
+  axiosInstance.interceptors.response.use(null, (error) => {
     const message = R.pathOr(error.message, ['response', 'data', 'error'], error)
-    dispatch({ type: appErrorCreate, error: { ...error, message } })
+    dispatch({ type: ServiceErrorActions.appErrorCreate, error: { ...error, message } })
```
Comment (minotogna) [inline webapp/app/appErrorsMiddleware.js:9]: here you should ```dispatch(ServiceErrorActions.createServiceError({error}))```

**Comment by minotogna** [comment]
Also, actions should be renamed using convention MODULE_TYPE

**Pull request merged**